draft-ietf-rats-architecture.md

```diff
@@ -1424,30 +1425,57 @@ in terms of `time(RX_v)-time(RG_v)`, then the Relying Party can check
 
 ## Example 3: Handle-based Passport Model Example
 
-Handles are a third option to establish time-keeping next to nonces or timestamps.
-Handles are opaque data intended to be available to all RATS roles that interact with each other, such as the Attester or Verifier, in specified intervals.
-To enable this availability, handles are distributed centrally by the Handle Distributor role over the network.
-As any other role, the Handle Distributor role can be taken on by a dedicated entity or collapsed with other roles, such as a Verifier.
-The use of handles can compensate for a lack of clocks or other sources of time on entities taking on RATS roles.
-The only entity that requires access to a source of time is the entity taking on the role of Handle Distributor.
+A third option to establish the epoch of Evidence or Attestation Result
+generation - next to nonces and timestamps ({{freshness}}) - is by using
+"handles".
 
-Handles are different from nonces as they can be used more than once and can be used by more than one entity at the same time.
-Handles are different from timestamps as they do not have to convey information about a point in time, but their reception creates that information.
-The reception of a handle is similar to the event that increments a relative tickcounter.
-Receipt of a new handle invalidates a previously received handle.
+Handles are opaque, non-predictable data regularly broadcast by a Handle Distributor.  
+Each handle represents a specific epoch,
+in which generated Evidence can be asserted to be fresh.
 
-In this example, Evidence generation based on received handles always uses the current (most recent) handle.
-As handles are distributed over the network, all involved entities receive a fresh handle at roughly the same time.
-Due to distribution over the network, there is some jitter with respect to the time the Handle is received, time(HR), for each involved entity.
-To compensate for this jitter, there is a small period of overlap (a specified offset) in which both a current handle and corresponding former handle are valid in Evidence appraisal: `validity-duration = time(HR'_v) + offset - time(HR_v)`. The offset is typically based on a network's round trip time.
-Analogously, the generation of valid Evidence is only possible, if the age of the handle used is lower than the validity-duration: `time(HR_v) - time(EG_a) < validity-duration`.
+The distribution of a new handle determines the transition from one epoch to
+the next.  The transition between epochs goes through an "overlap" phase in
+which the previous epoch is still extant while the current epoch is
+established.  The span of the overlap phase is determined solely by the
+transmission delay of the underlying communication channel,
+`time(HR)-time(HT)`.  Once the overlap phase is over, an epoch is fully
```

Comment:
I don't think this is true, since the receiver has no idea when time(HT) is.   To avoid race conditions, receivers (Verifiers / Relying Parties) need to accept both the previous handle and the current handle.  In that case, the original text inequalities are correct and this one is not.

Reply:
> I don't think this is true, since the receiver has no idea when time(HT) is. 

I see your point, but I wouldn't say it is a false statement.  It is however both imprecise (it should be the worst case delay among all receivers) and not really useful from an operational point of view.  Therefore I'm happy to remove it from the sentence which has clear enough prose to express the idea already.

I think we also need to highlight that the extent of the overlap, the thing previously known as "offset" (would "pad" be a better name?), is a central protocol parameter which needs calibration and agreement among all the participants to be meaningful.  It's not a unilateral decision that a protocol participant makes up by itself.

And another related thing (which I just realised, sorry for overloading the comment) is that the discussion seems to be assuming a network topology where V is _closer_ to the Handle Distributor than V - and therefore V’s epoch starts _before_ A’s epoch.  We should also consider the dual situation and therefore the case where Evidence arrives at V with a handle that V has not yet received.  This calls for extending the “offset” on the left to buffer _maybe valid_ handles?

> [...] the original text inequalities are correct and this one is not

One of the troubles I was having with the original text is that I really don’t understand what the inequality is supposed to say.  In particular, how is `time(HR_v)-time(EG_a)` a definition of “age of the handle used”?  Besides, it seems that there is an unstated assumption here that A and V have synchronised clocks - otherwise time(EG_a) and time(HR_v) are not on the same timeline and can't be meaningfully combined.

Reply:
First of all, you intended to write "where V is _closer_ to the Handle Distributor than A", I think.

Additionally, I think we came to the conclusion that it is preferable to use an event (defined in the table) in the text of the Appendix in order to provide better context. Simply removing the sentence with time(HT) at the end would go against that. Hence, I would recommend to not simply remove it, but to replace it with a paragraph that highlights why event types, such as time(HT), that can frame the extent of an overlap phase are vital to readers of the document here.

In general, Handle Distribution is in support of various scenarios. In some of them the receivers have no clocks at all, in some of them only a few - e.g. the Verifier. Maybe it would help with the scenario description, if we add a Clock to the Verfier? (so that only the swarm of Attesters has no clock)

Reply:
> First of all, you intended to write "where V is _closer_ to the Handle Distributor than A", I think.

Your thinking is correct, fixed.  Thanks for spotting it!

> Additionally, I think we came to the conclusion that it is preferable to use an event (defined in the table) in the text of the Appendix in order to provide better context. Simply removing the sentence with time(HT) at the end would go against that. Hence, I would recommend to not simply remove it, but to replace it with a paragraph that highlights why event types, such as time(HT), that can frame the extent of an overlap phase are vital to readers of the document here.

I agree.  Frankly, I care more about clarity at the conceptual level rather than anything else.  So, whatever device helps to better grasp the ideas is very welcome.

> In general, Handle Distribution is in support of various scenarios. In some of them the receivers have no clocks at all, in some of them only a few - e.g. the Verifier. Maybe it would help with the scenario description, if we add a Clock to the Verfier? (so that only the swarm of Attesters has no clock)

What I was trying to say is just that the second inequality makes no sense because the two `time()` functions apply to different timelines (and are therefore the comparison is between pears and apples!) unless the underlying clocks at V and A are synchronised.  If we cannot assume the presence of clocks at nodes implementing the A role, let alone synchronised clocks, well that's another good reason for dropping the inequality, isn't it?

If instead there is at least some assumption on clock presence in some of the roles (and possibly a synchronisation requirement?), that should be clearly stated.